### PR TITLE
Tabs for the new Display Package page redesign 

### PR DESCRIPTION
**Pull request by sophiamfavro**
## Solution 
I used bootstrap to created tabs to match the new redesign. The tabs were made sticky the same way that the installation tabs are by adding code in the .js file that saves the last tab clicked onto local storage. For styling, I went into the .less file and added the blue bar at the bottom of the clicked on tab and added padding to keep the distance of the tab content to the tabs consistent for each content. The final now looks like this: 
image

### Mobile
The mobile is still a work in progress as we are finalizing how we want it to look to keep the experience just as strong as it will be on desktop. All the features and experiences are the same currently and the only issue is the tabs in the bottom row get pushed if a tab in the top row is clicked(see picture below). We decided to punt this as we have a work item(https://github.com/nuget/nugetgallery/issues/8611) for mobile and want to wait until the redesign is finalized to make changes.
image
image

## Testing
To test my solution I tested all the links on both the new page and old page to make sure that it all worked correctly. I then inspected the page and looked at the console to make sure that there were no errors in the .js file. I also double checked that the sticky tabs worked by reloading the page, going to a different package, and seeing that the correct value was being stored in local storage. Finally, to make sure that the page worked on deleted and unlisted packages I deleted and unlisted packages in my local enviroment. I then looked at them to make sure that only the used by and versions tabs were shown to match what we currently do on the old page. 
If you have any questions feel free to reach out

### CODE REVIEW

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 1]
```diff
@@ -324,4 +324,36 @@
...
+
+    h1:first-child,
+    h2:first-child,
+    h3:first-child,
+    h4:first-child,
+    h5:first-child,
+    h6:first-child
+    {
+        margin-top:0px;
+    }
```
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-display-package-v2.less:357]: Just to be consistent in style:

```suggestion
    h6:first-child {
        margin-top: 0px;
    }
```

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 2]
```diff
@@ -324,4 +324,36 @@
...
+      border-right: 0px;
+      border-top: 0px;
+      font-size: 14px;
+      font-family: @font-family-base;
+      color: #323130;
+    }
+  }
+
+  .body-tab-content {
+    padding-top:30px;
```
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-display-package-v2.less:347]:
```suggestion
    padding-top: 30px;
```

File: src/NuGetGallery/Scripts/gallery/page-display-package-v2.js [hunk 3]
```diff
@@ -125,30 +86,36 @@
...
+            storage.setItem(installationKey, e.target.id);
         });
-    }
 
-    if (window.nuget.isGaAvailable()) {
-        // Emit a Google Analytics event when the user expands or collapses the Dependencies section.
-        $("#dependency-groups").on('hide.bs.collapse show.bs.collapse', function (e) {
-            ga('send', 'event', 'dependencies', e.type);
-        });
+        // set perferred body tab
```
Comment (loic-sharma) [inline src/NuGetGallery/Scripts/gallery/page-display-package-v2.js:102]:
```suggestion
        // set preferred body tab 
```

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 4]
```diff
@@ -501,374 +496,350 @@
...
+                            </div>
+                        </div>
+                        <div role="tabpanel" class="tab-pane" id="releasenotes-tab">
+                            @if(!String.IsNullOrWhiteSpace(Model.ReleaseNotes)) 
+                            { 
+                                <p>@Html.PreFormattedText(Model.ReleaseNotes, Config)</p>
+                            } 
+                        </div>
                     </div>
                 </div>
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:841]: Should we not include the `div` for release notes if there are no release notes? Something like:

```suggestion
                        @if (!String.IsNullOrWhiteSpace(Model.ReleaseNotes)) 
                        {
                            <div role="tabpanel" class="tab-pane" id="releasenotes-tab"> 
                                <p>@Html.PreFormattedText(Model.ReleaseNotes, Config)</p>
                            </div>
                        }
                    </div>
                </div>
```

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 5]
```diff
@@ -268,69 +268,64 @@
...
-
                 @if (Model.Validating)
                 {
                     @ViewHelpers.AlertWarning(
-                        @<text>
-                            <strong>This package has not been published yet.</strong> It will appear in search results and
-                            will be available for install/restore after both validation and indexing are complete. Package
-                            validation and indexing may take up to an hour. <a href="https://aka.ms/NuGetPackageValidation">Read more</a>.
-                        </text>
+                            @<text>
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:274]: There's a lot of whitespace changes again. Ugh auto-formatting 😢

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 6]
```diff
@@ -501,374 +496,350 @@
...
-                        if (Model.Dependencies.DependencySets == null)
-                        {
-                            if (Model.CanDisplayPrivateMetadata)
+                    </ul>
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-sm-9">
+                    <div class="tab-content body-tab-content">
+                        <div role="tabpanel" class="tab-pane active" id="readme-tab">
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:518]: For all of the tabs that may not be shown, we should skip both the tab pane's div as well as its content. For example, instead of:

```cshtml
<div role="tabpanel" class="tab-pane" ...>
  @if (should show this pane)
  {
    // Content...
  }
</div>
```

We should do this instead:

```cshtml
@if (should show this pane)
{
  <div role="tabpanel" class="tab-pane" ...>
    // Content...
  </div>
}
```

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 7]
```diff
@@ -501,374 +496,350 @@
...
                                 {
-                                    <h3>
-                                        <strong>NuGet packages</strong>
-                                    </h3>
-                                    <p>
-                                        This package is not used by any NuGet packages.
-                                    </p>
+                                    <p>@Html.PreFormattedText(Model.Description, Config)</p>
                                 }
                             }
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:541]: You can merge the `else` and `if` here:

```cshtml
else if (!Model.Deleted && !String.IsNullOrWhiteSpace(Model.Description))
{
  ...
}
```

File: src/Bootstrap/less/theme/page-display-package-v2.less (hunk 1 above)
Comment (sophiamfavro) [reply src/Bootstrap/less/theme/page-display-package-v2.less:357]: Done :)

File: src/Bootstrap/less/theme/page-display-package-v2.less (hunk 2 above)
Comment (sophiamfavro) [reply src/Bootstrap/less/theme/page-display-package-v2.less:347]: done 👍

File: src/NuGetGallery/Scripts/gallery/page-display-package-v2.js (hunk 3 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Scripts/gallery/page-display-package-v2.js:102]: done 👍

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml (hunk 4 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:841]: done 👍

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml (hunk 5 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:274]: Done 👍

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml (hunk 6 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:518]: Done 👍

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml (hunk 7 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:541]: done 👍

File: src/NuGetGallery/Scripts/gallery/page-display-package-v2.js [hunk 8]
```diff
@@ -125,30 +86,36 @@
 
     var storage = window['localStorage'];
     if (storage) {
-        var key = 'preferred_tab';
+        var installationKey = 'preferred_tab';
```
Comment (zhhyu) [inline src/NuGetGallery/Scripts/gallery/page-display-package-v2.js:89]: very nit: Add a comment like "body tab":
// set preferred installation tab

File: src/Bootstrap/dist/css/bootstrap-theme.css [hunk 9]
```diff
@@ -334,6 +334,11 @@ img.reserved-indicator-icon {
 .sortable {
   cursor: pointer;
 }
+@media (max-width: 767.9px) {
+  .hidden-xs {
+    display: none !important;
```
Comment (zhhyu) [inline src/Bootstrap/dist/css/bootstrap-theme.css:339]: Is it expected that we introduce this change? My apologies that I am not a less/css expert.

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 10]
```diff
@@ -321,7 +316,8 @@
                             {
                                 <text>Please contact <a href="mailto:support@nuget.org">support@nuget.org</a> to help fix your package.</text>
                             }
-                        </text>)
+                            </text>
```
Comment (zhhyu) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:319]: Additional spaces?

**Comment by zhhyu** [comment]
I think we may need to polish the accessibility later, but it's out of scope of this PR.

**Review by zhhyu — APPROVED** [review]

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml (hunk 10 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:319]: Yes, Thank you for catching this! I found it and the spacing was definitely off

File: src/NuGetGallery/Scripts/gallery/page-display-package-v2.js (hunk 8 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Scripts/gallery/page-display-package-v2.js:89]: Done 👍 Thank you for the suggestion

File: src/Bootstrap/dist/css/bootstrap-theme.css (hunk 9 above)
Comment (sophiamfavro) [reply src/Bootstrap/dist/css/bootstrap-theme.css:339]: This file is auto-generated by grunt. To be honest, it is not clear why grunt chose to add this now, but it seems to be okay regardless. I tried deleting the file and running grunt again to recreate it and it gets added every time

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 11]
```diff
@@ -324,4 +324,35 @@
...
+  .body-tab-content {
+    padding-top: 30px;
+
+    h1:first-child,
+    h2:first-child,
+    h3:first-child,
+    h4:first-child,
+    h5:first-child,
+    h6:first-child {
+        margin-top:0px;
```
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-display-package-v2.less:355]: It looks like we use 2 spaces for indentation on this file:

```suggestion
      margin-top: 0px;
```

File: src/NuGetGallery/Scripts/gallery/page-display-package-v2.js [hunk 12]
```diff
@@ -125,30 +86,37 @@
 
     var storage = window['localStorage'];
     if (storage) {
-        var key = 'preferred_tab';
+        // set preferred installation instruction tab
```
Comment (loic-sharma) [inline src/NuGetGallery/Scripts/gallery/page-display-package-v2.js:89]: Hey on line 82-85 this file has:

```js
    $("#undo-pending-edits").click(function (e) {
        e.preventDefault();
        $(this).closest('form').submit();
    })
```

This is dead code (a long time ago we used to support editing packages, but we've removed that). Let's delete those lines since they're not doing anything.

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 13]
```diff
@@ -501,374 +498,350 @@
...
-                                       aria-hidden="true"></i>
-                                    <span>Dependencies</span>
-                                </a>
-                            </h2>
-
-                            if (Model.Dependencies.DependencySets.Any())
-                            {
-                                <ul class="list-unstyled panel-collapse collapse dependency-groups" id="dependency-groups">
-                                    @foreach (var dependencySet in Model.Dependencies.DependencySets)
+                            <div role="tabpanel" class="tab-pane" id="dependencies-tab">
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:544]: We should skip this tabpanel entirely if the package is deleted. See: https://github.com/NuGet/NuGetGallery/pull/8625#discussion_r648491782

File: src/NuGetGallery/App_Data/Files/Content/flags.json [hunk 14]
```diff
@@ -26,7 +26,8 @@
     "NuGetGallery.ManagePackagesVulnerabilities": "Enabled",
     "NuGetGallery.DisplayFuGetLinks": "Enabled",
     "NuGetGallery.PatternSetTfmHeuristics": "Enabled",
-    "NuGetGallery.EmbeddedIcons": "Enabled"
+    "NuGetGallery.EmbeddedIcons": "Enabled",
+    "NuGetGallery.MarkdigMdRendering": "Enabled"
```
Comment (loic-sharma) [inline src/NuGetGallery/App_Data/Files/Content/flags.json:30]: Currently git thinks there's a git conflict since @lyndaidaii also fixed this in https://github.com/NuGet/NuGetGallery/pull/8630. You can undo these changes before you merge to unconfuse git  :)

**Review by loic-sharma — APPROVED** [review]
Great work! 🚀

File: src/Bootstrap/less/theme/page-display-package-v2.less (hunk 11 above)
Comment (sophiamfavro) [reply src/Bootstrap/less/theme/page-display-package-v2.less:355]: done 👍

File: src/NuGetGallery/Scripts/gallery/page-display-package-v2.js (hunk 12 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Scripts/gallery/page-display-package-v2.js:89]: done 👍

File: src/NuGetGallery/App_Data/Files/Content/flags.json (hunk 14 above)
Comment (sophiamfavro) [reply src/NuGetGallery/App_Data/Files/Content/flags.json:30]: done 👍
